mobius-coroutines/src/main/java/com/spotify/mobius/coroutines/CoroutinesSubtypeEffectHandlerBuilder.kt

```diff
@@ -252,7 +258,8 @@ class CoroutinesSubtypeEffectHandlerBuilder<F : Any, E : Any> {
             override fun dispose() {
                 scope.cancel("Effect Handler disposed")
                 eventsChannel.close()
-                subEffectChannels.forEachValue(1) { it.close() }
+                subEffectChannelsMap.values.forEach { it.close() }
```

Comment:
Should this be protected by the mutex so as to avoid a race where an `accept` might add a new channel to the map? I'm not sure what the guarantees of 'scope.cancel' are, but I'm thinking something like this:

1. accept() called, mutex.withLock() enters on physical thread 1
2. dispose() called, scope cancelled on physical thread 2
3. all current channels in map are closed on physical thread 2
4. physical thread 1 adds an effect handler to the map and then gets cancelled.

Reply:
I missed this one. I though `scope.cancel()` blocked until all jobs where cancelled. But in reality it sends the cancellation signal and continues processing. 

I will have to use `runBlocking` in order to use the `Mutex` since `dispose` is not a suspend function.

I'm also adding an early cancellation when the channels are created checking for `isActive` to avoid the process of  creating an effectChannel when the scope is not longer active.